### PR TITLE
pkg/system: remove Umask() utility

### PR DESCRIPTION
**Pull request by thaJeztah**
It was only used in a couple of places, and in most places shouldn't be used as those locations were in unix/linux-only files, so didn't need the wrapper.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]

File: pkg/archive/diff_unix.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//go:build !windows
+// +build !windows
+
+package archive
+
+import "golang.org/x/sys/unix"
+
+// overrideUmask sets current process's file mode creation mask to newmask
+// and returns a function to restore it.
```
Comment (corhere) [inline pkg/archive/diff_unix.go:9]: Document why this function is dangerous in a multithreaded process to dissuade anyone from thinking it is possible to safely change and restore the umask in Go programs.

**Review by corhere — COMMENTED** [review]
The refactor itself LGTM, but I worry people might stumble across the code and think it's safe to muck with a Go program's umask arbitrarily.

**Comment by thaJeztah** [comment]
> people might stumble across the code and think it's safe to muck with a Go program's umask arbitrarily.

At least it's un-exported now, so less tempting to use it 😄 

I'll add a blurb to the GoDoc

File: pkg/archive/archive_linux_test.go [hunk 2]
```diff
@@ -86,9 +86,8 @@ func checkFileMode(t *testing.T, path string, perm os.FileMode) {
 }
 
 func TestOverlayTarUntar(t *testing.T) {
-	oldmask, err := system.Umask(0)
-	assert.NilError(t, err)
-	defer system.Umask(oldmask)
+	oldmask := unix.Umask(0)
+	defer unix.Umask(oldmask)
```
Comment (thaJeztah) [inline pkg/archive/archive_linux_test.go:90]: Hm.. these are in the same package; I should probably use `overrideUmask()` here (which also leads to the warning I'm adding).

**Comment by thaJeztah** [comment]
@corhere updated; PTAL

**Review by corhere — APPROVED** [review]

**Pull request merged**